### PR TITLE
Swift wrapper

### PR DESCRIPTION
**Pull request by LucFauvel**

### CODE REVIEW

File: src/webauthn/authenticator/mod.rs [hunk 1]
```diff
@@ -157,7 +205,7 @@ impl WebauthnAuthenticator {
                 )
             }
 
-            CoseAlgorithmIdentifier::EC2 => {
+            CoseAlgorithmIdentifier::ES256 => {
```
Comment (jo-trep) [inline src/webauthn/authenticator/mod.rs:208]: we don't need EC2 anymore?

File: src/webauthn/authenticator/native.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        ptr::null_mut,
+    };
+    use uuid::Uuid;
+
+    pub struct AuthenticatorCreationResponse {
+        pub private_key_response: String,
+        pub attestation_object: Vec<u8>,
+    }
+
+    pub struct AuthenticatorRequestResponse {
```
Comment (Samuel-B-D) [inline src/webauthn/authenticator/native.rs:22]: Same here, passed across FFI boundaries.
Rust ABI is not stable, so anything you pass across FFI boundaries should be `#[repr(C)]`.
https://doc.rust-lang.org/nomicon/other-reprs.html#reprc

It is passed here:
- https://github.com/Devolutions/slauth/pull/80/files#diff-d1734eb72d2e1cebe69d5533ae1d7d92372e8babd1b70cabfd11472ca526237dR125
- https://github.com/Devolutions/slauth/pull/80/files#diff-d1734eb72d2e1cebe69d5533ae1d7d92372e8babd1b70cabfd11472ca526237dR155
- https://github.com/Devolutions/slauth/pull/80/files#diff-d1734eb72d2e1cebe69d5533ae1d7d92372e8babd1b70cabfd11472ca526237dR167

**Review by jo-trep — APPROVED** [review]

File: src/webauthn/authenticator/native.rs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        },
+    };
+    use std::{
+        ffi::{c_int, c_uchar, CString},
+        os::raw::c_char,
+        ptr::null_mut,
+    };
+    use uuid::Uuid;
+
+    pub struct AuthenticatorCreationResponse {
```
Comment (Samuel-B-D) [inline src/webauthn/authenticator/native.rs:17]: Isn't this a struct that you pass around across FFI boundaries?
Rust ABI is not stable, so anything you pass across FFI boundaries should be `#[repr(C)]`.
https://doc.rust-lang.org/nomicon/other-reprs.html#reprc

Edit: it is passed here:
- https://github.com/Devolutions/slauth/pull/80/files#diff-d1734eb72d2e1cebe69d5533ae1d7d92372e8babd1b70cabfd11472ca526237dR34
- https://github.com/Devolutions/slauth/pull/80/files#diff-d1734eb72d2e1cebe69d5533ae1d7d92372e8babd1b70cabfd11472ca526237dR47

File: src/webauthn/authenticator/native.rs (hunk 2 above)
Comment (Samuel-B-D) [reply src/webauthn/authenticator/native.rs:22]: As long as it's just used as an OpaquePointer and never manipulated outside, it's fine.

File: src/webauthn/authenticator/native.rs (hunk 3 above)
Comment (Samuel-B-D) [reply src/webauthn/authenticator/native.rs:17]: As long as it's just used as an OpaquePointer and never manipulated outside, it's fine.

**Pull request merged**